In [3]:
%run "../../../common/0_notebooks_base_setup.py"    

/media/paulati/Nuevo vol/paula/dh/2021/dsad_2021_paula/common
default checking
Running command `conda list`... ok
jupyterlab=2.2.6 already installed
pandas=1.1.5 already installed
bokeh=2.2.3 already installed
seaborn=0.11.0 already installed
matplotlib=3.3.2 already installed
ipywidgets=7.5.1 already installed
pytest=6.2.1 already installed
chardet=4.0.0 already installed
psutil=5.7.2 already installed
scipy=1.5.2 already installed
statsmodels=0.12.1 already installed
scikit-learn=0.23.2 already installed
xlrd=2.0.1 already installed
Running command `conda install --yes nltk=3.5.0`... ok
Solving environment: ...working... done

# All requested packages already installed.


unidecode=1.1.1 already installed
pydotplus=2.0.2 already installed
pandas-datareader=0.9.0 already installed
flask=1.1.2 already installed


---

<img src='../../../common/logo_DH.png' align='left' width=35%/>


# Estadística Inferencial

## Imports

In [4]:
import scipy.stats as stats
import pandas as pd
import numpy as np
import math
import seaborn as sns
import matplotlib.pyplot as plt

## Dataset

Utilizamos una muestra aleatoria de puntajes de influencia de redes sociales del servicio klout. 

Klout era un sitio web y una aplicación móvil que utilizaba análisis de redes sociales para calificar usuarios de acuerdo con su influencia social en línea. 

El sitio calculaba un "Klout Score", que era un valor numérico entre 1 y 100 donde puntuaciones más altas correspondian a una mayor "influencia social" en línea.

De una población de más de 620 millones de puntajes obtuvimos una muestra de tamaño 1048

Pueden ver más detalles sobre klout <a href="https://topinfluencers.es/que-es-klout/" target="_blank">aquí</a>

## Leemos los datos

In [5]:
data_klout = pd.read_csv('../Data/klout-scores.csv', header=None, names=['scores'])
data_klout.head()

scores
0  25.104179
1  60.908747
2  23.222238
3  51.553491
4  22.629690

## Ejercicio 1: Intervalos de confianza

Escribir una función que dado los datos como una instancia de Series, un número n_sample, y un valor alpha tome una muestra de los datos de tamaño n y calcule la media muestral y el intervalo de confianza para la media a partir de esa muestra.

$$\bar{x}-z_{\alpha/2}\cdot\frac{\sigma}{\sqrt{n}}\lt\mu_{estimator}\lt\bar{x}+z_{\alpha/2}\cdot\frac{\sigma}{\sqrt{n}}$$ 

Usemos esa función para calcular los intervalos de confianza de nivel 95% para las medias de los scores de data_klout en una muestra de tamaño 20 y otra de tamaño 100.


## Ejercicio 2: Normal vs. T de Student

**2.1** Grafiquemos los datos de klout-scores para tenes alguna intuición sobre su distribución.

**2.2** En el ejercicio anterior calculamos los intervalos de confianza de nivel 95% para las medias de los scores de data_klout en una muestra de tamaño 20 y otra de tamaño 100 usando el desvío estándar de la muestra como aproximación a sigma, y distribución normal. 

Repitamos ahora este cálculo para valores de n entre 5 y 200 con paso 5 (o sea 5, 10, 15, .... 200)

Y calculemos para esos mismos valores de n las longitudes de los intervalos de confianza considerando que **no conocemos el desvío estándar poblacional**

**2.3** Construyamos un dataset cuyas columnas sean 

* tamaño del intervalo de confianza de los resultado del ejercicio 1 o 2

* n, 5 a 200 con paso 5 

* fuente, si el resultado proviene del ejercicio 1 (desvío estándar poblacional conocido) o del ejercicio 2 (desvío estándar poblacional desconocido)

**2.4** A partir de este dataset grafiquemos las dos series de resultados en función de n. 

**2.5** ¿A partir de qué valor de n el cálculo del ejercicio 1 es equivalente al del ejercicio 2? Para responder esta pregunta grafiquemos para cada valor de n el valor absoluto de la diferencia entre los resultados de los ejercicios 1 y 2.



---

**2.1**. Graficamos los datos de klout-scores

---

**2.2**

---

**2.3**

---

**2.4**



---

**2.5**